int16[] q_matrix_row